client/python/openlineage/client/transport/amazon_datazone.py

```diff
@@ -0,0 +1,70 @@
+# Copyright 2018-2025 contributors to the OpenLineage project
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+
+import logging
+from dataclasses import dataclass
+from typing import TYPE_CHECKING, Any
+
+from openlineage.client import event_v2
+from openlineage.client.run import RunEvent
+from openlineage.client.serde import Serde
+from openlineage.client.transport.transport import Config, Transport
+
+if TYPE_CHECKING:
+    from openlineage.client.client import Event
+
+log = logging.getLogger(__name__)
+
+@dataclass
+class AmazonDataZoneConfig(Config):
+    domain_id: str
+    endpoint_override: str | None = None
+
+    @classmethod
+    def from_dict(cls, params: dict[str, Any]) -> AmazonDataZoneConfig:
+        if "domainId" not in params:
+            msg = "`domainId` key not passed to AmazonDataZoneConfig"
+            raise RuntimeError(msg)
+        return cls(domain_id=params["domainId"], endpoint_override=params.get("endpointOverride", None))
+
+
+class AmazonDataZoneTransport(Transport):
+    kind = "amazon_datazone_api"
+    config_class = AmazonDataZoneConfig
+
+    def __init__(self, config: AmazonDataZoneConfig) -> None:
+        self.config = config
+        self._setup_datazone(self.config.endpoint_override)
+
```

Comment:
```suggestion
log.debug(
            "Constructing OpenLineage transport that will send events "
            "to Amazon Datazone domain `%s`.",
            config.domain_id,
        )
```
Can we add something like this? It's present in all other transports, and can help with debugging.